colossalai/auto_parallel/tensor_shard/utils/misc.py

```diff
@@ -29,3 +31,37 @@ def wrapper(*args, **kwargs):
             return None
 
     return wrapper
+
+
+def check_sharding_spec_validity(sharding_spec: ShardingSpec, tensor: torch.Tensor):
```

Comment:
This function is moved from `tests/test_auto_parallel/test_tensor_shard/test_node_handler/common.py` so that it can be used by the node handler directly.